tasks/sanitize-checks.yml

```
@@ -0,0 +1,9 @@
+- name: Defend against defined but null datadog_checks variable
+  set_fact:
+      datadog_checks: "{{datadog_checks | default({})}}"
+  when: datadog_checks is none
+
+- name: Resolve datadog_tracked_checks
+  set_fact:
+    datadog_tracked_checks: "{{ datadog_checks | list + datadog_additional_checks }}"
```

To be extra safe, we could also guard against `datadog_additional_checks` being set to None / empty values:
```suggestion
    datadog_tracked_checks: "{{ datadog_checks | list + datadog_additional_checks | default([], true) }}"
```

And, since we're defining it here, I'd remove the `datadog_tracked_checks` definition from the `defaults/main.yml` file